tests/TestCaseBase.php

```diff
@@ -0,0 +1,306 @@
+<?php
+/**
+ * Parser Reflection API
+ *
+ * @copyright Copyright 2016, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+namespace Go\ParserReflection;
+
+class TestCaseBase extends \PHPUnit_Framework_TestCase
+{
+
+    /**
+     * Extracts namespace from class, function or constant name.
+     *
+     * @return string
+     */
+    protected function getNamespaceFromName($name)
+    {
+        $nameParts = explode('\\', $name);
+        array_pop($nameParts);
+        return implode('\\', $nameParts);
+    }
+
+    /**
+     * Extracts short name from class, function or constant name.
+     *
+     * @return string
+     */
+    protected function getShortNameFromName($name)
+    {
+        $nameParts = explode('\\', $name);
+        return array_pop($nameParts);
+    }
+
+    protected function getStringificationOf($value, $maxLen = 128)
```

Comment:
Wow! Why do we need such a complex function? I'd vote to drop it and replace with something more general, without dark magic inside...

Reply:
This method is **most** of why this PR is still `[WIP]` right now... it is there exclusively for test failure readability, so I think it adds minimal risk, but needs its own unit tests, and should be moved into a class in `tests/Support`, or somewhere similar.

Opinions?

Reply:
Let me amend that.... I intend to use this function for future tests, not just test failure reporting, but it is **currently** only used for test error reporting.

Reply:
@lisachenko, I did some exploring and found `PHPUnit` uses `SebastianBergmann\Exporter\Exporter` to avoid this very mess. I see no reason I can't do similarly. I'm (slowly) working on extracting a PR from this one that refactors the test suite and will address this there.